app/src/main/java/com/hover/stax/accounts/AccountsViewModel.kt

```diff
@@ -2,26 +2,30 @@ package com.hover.stax.accounts
 
 import android.app.Application
 import android.content.Context
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.MediatorLiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.*
 import com.hover.sdk.actions.HoverAction
+import com.hover.sdk.sims.SimInfo
 import com.hover.stax.R
 import com.hover.stax.actions.ActionRepo
 import com.hover.stax.data.local.bonus.BonusRepo
 import com.hover.stax.data.local.accounts.AccountRepo
 import com.hover.stax.domain.model.Account
 import com.hover.stax.domain.model.PLACEHOLDER
+import com.hover.stax.domain.use_case.sims.GetLivePresentSimUseCase
 import com.hover.stax.schedules.Schedule
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.receiveAsFlow
 import kotlinx.coroutines.launch
 
-class AccountsViewModel(application: Application, val repo: AccountRepo, val actionRepo: ActionRepo, private val bonusRepo: BonusRepo) : AndroidViewModel(application),
+class AccountsViewModel(application: Application,
```

Comment:
All Sim functions should move to their own `ViewModel`. We're moving from function-based ViewModels to screen-based ViewModels. It's easier to track actions and side-effects that way. Use use-cases to compose your functionality as they are reusable.

Reply:
Thanks for this. Code updated